internal/scanner/scanner.go

```diff
@@ -231,7 +229,7 @@ func (s *Scanner) ScanAllNamespaces(ctx context.Context, runUID string) error {
 			defer workers.Done()
 
 			if e := s.ScanNamespace(ctx, namespaceName, runUID); e != nil {
-				s.logger.ErrorContext(ctx, "error scanning namespace", slog.String("error", err.Error()), slog.String("ns", namespaceName))
+				s.logger.ErrorContext(ctx, "error scanning namespace", slog.String("error", e.Error()), slog.String("ns", namespaceName))
 				err = errors.Join(err, e)
```

Comment:
@kubewarden/kubewarden-developers, this changes fix another bug that I've found while I test this PR. The `ScanAllNamespaces` was not returning the error when a namespace scan fail. I've added an unit test to avoid a regression here.